diff --git a/src/webserver/civetweb/mod_lua.inl b/src/webserver/civetweb/mod_lua.inl
index 26f281ee..e9a13835 100644
--- a/src/webserver/civetweb/mod_lua.inl
+++ b/src/webserver/civetweb/mod_lua.inl
@@ -3208,8 +3208,8 @@ handle_lsp_request(struct mg_connection *conn,
 	 *
 	 */
	run_lsp = run_lsp_civetweb;
-	if ((addr[0] == '<') && (addr[1] != '?')) {
+	if ((addr[0] != 'H')) { // Pi-hole change: Check if file starts with "H" for HTTP
		run_lsp = run_lsp_kepler;
	}
 
 	/* We're not sending HTTP headers here, Lua page must do it. */
